schema.md

```diff
@@ -158,11 +161,64 @@ same database, the duplicate entry could be written using only the `id`,
 The `related` field gives a list of IDs of closely related vulnerabilities, such
 as the same problem in alternate ecosystems. 
 
-### package field
+### summary, details fields
+
+The `summary` field gives a one-line, English textual summary of the
+vulnerability. It is recommended that this field be kept short, on the order of
+no more than 120 characters.
+
+The `details` field gives additional English textual details about the
+vulnerability. The field is plain text. Newline characters must be considered
+line breaks when displaying the text, but the display need not use a fixed-width
+font, and the text should not assume one.
+
+The `summary` field is plain text.
+
+The `details` field is CommonMark markdown (a subset of GitHub-Flavored
+Markdown). Display code may at its discretion sanitize the input further, such
+as stripping raw HTML and links that do not start with http:// or https://.
+Databases are encouraged not to include those in the first place. (The goal is
+to balance flexibility of presentation with not exposing vulnerability database
+display sites to unnecessary vulnerabilities.)
+
+### affected fields
+
+The `affected` field is a JSON array containing objects that describes the
+affected package versions, meaning those that contain the vulnerability.
+
+Within each object in the `affected` array, the `package` field identifies the
+package containing the vulnerability. In most cases, there should be exactly one
+entry in the `affected` array per affected `package` to describe all affected
+versions. In rare cases, for example if the `ecosystem_specific` encodes
+platform information that doesn't apply equally to all listed versions and
+ranges, a separate entry with the same `package` in the `affected` array may be
+needed.
+
+The `versions` field can enumerate a specific set of affected versions, and the
+`ranges` field can list ranges of affected versions, under a given defined
+ordering. **A version is considered affected if it lies within any one of the
+ranges or is listed in the versions list.**
+
+The `versions` list should - with one exception - always be present, to allow
+software to answer the question "is this specific version affected?" without
+having to contain code specific to every different ecosystem. The one exception
+is if the affected versions are valid SemVer 2.0 versions which can be
+accurately summarized by one or more non-overlapping SemVer ranges. In that
```

Comment:
@oliverchang Singling out semver as being a perfect version scheme assumes that humans handling semver versions will never break semver. I think that this cannot hold at scale. 
 IMHO there is no reason to treat SEMVER specially and I explained some rationale in the related https://github.com/CVEProject/cve-schema/issues/87#issuecomment-904398866

Instead a version range may best treated as what it is: a hint for tools and humans to help form a proper `versions` enumeration.

Reply:
Thanks for the feedback @pombredanne. I agree an explicit list of versions makes this unambiguous in all cases.

I think tooling/validators can help with making sure the SEMVER values are valid. One other reason why we need this is that in ecosystems like Go (which use/enforce Semver, it's infeasible to enumerate every possible version. Every commit can be a "pseudoversion" (valid semver) in Go.

Reply:
Thank you for the clarification. This makes sense now. I had not internalized this aspect of Go modules versioning. That's actually pretty sleek and clean. It creates some extra work when trying to enumerate these when compared to other package types... but that's an issue for tools/db to deal with..!